cmd/git-bundle-web-server/bundle-server.go

```diff
@@ -107,6 +113,13 @@ func (b *bundleWebServer) serve(w http.ResponseWriter, r *http.Request) {
 
 	route := owner + "/" + repo
 
+	if b.authorize != nil {
+		authResult := b.authorize(r, owner, repo)
+		if authResult.WriteResponse(w) {
+			return
+		}
```

Comment:
I originally had it as the latter, but I wasn't thrilled with needing to re-check `Authorized()` in `WriteResponse()` (because if I was doing that anyway, why even have the original `Authorized()` check?). 

I'll add a comment, but I'll also try to come up with a better name (`AuthorizeOrWriteResponse()`? or maybe something simpler like `ApplyResult()`).

Reply:
If we wanted to give auth middleware full control of the pipeline then really we should really only expect a boolean "return bundles or exit?" and have the middleware handle the entire response?

```go
if b.authorize != nil {
    if !b.authorize(r, w, owner, repo) {
        return
    }
}
```

I assume the only expected front-end/consumer of the bundle server endpoints is Git (never a browser etc)?

Reply:
It took a few iterations, but I've made the `AuthResult`s a bit more generic (where `Allow()` == "apply headers & serve content" and `Deny()` == "apply headers & respond with given 4XX code"). I'd like to keep the middleware at least a bit sandboxed (it helps keep the bundle server aligned with what Git expects/supports and IMO makes plugin writing simpler), but if there's anything else that might be valuable to add to the current interface, I'm happy to update!